### PR TITLE
Add e2e for connection reset issue

### PR DESCRIPTION
**Pull request by anfernee**
Regression test for #74839

### CODE REVIEW

**Comment by anfernee** [comment]
@MrHohn

File: test/e2e/network/networking.go [hunk 1]
```diff
@@ -57,6 +57,10 @@ var _ = SIGDescribe("Networking", func() {
 			framework.CheckConnectivityToHost(f, "", "ping-test", "google.com", framework.IPv6PingCommand, 30))
 	})
 
+	It("should resolve connrection reset issue in #74839 [Feature:Regression]", func() {
```
[the remark this answers is not in the document]
Comment (MrHohn) [reply test/e2e/network/networking.go]: And can we make a better description/title for this? Don't think mentioning issue number in title is a good practice :)

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -4576,6 +4576,122 @@ func CheckConnectivityToHost(f *Framework, nodeName, podName, host string, pingC
 	return err
 }
 
+// Regression test for #74839
+// https://github.com/kubernetes/kubernetes/issues/74839
+func CheckConnectionReset(f *Framework) error {
```
Comment (MrHohn) [inline test/e2e/framework/util.go]: This doesn't seem like a common enough thing to put in framework/util. Maybe put these back to the actual test file?

Comment (MrHohn) [reply test/e2e/framework/util.go]: Also would be great to have a high level description on what this test does --- as it seems to be different than what we usually test?

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -4576,6 +4576,122 @@ func CheckConnectivityToHost(f *Framework, nodeName, podName, host string, pingC
...
+								MatchLabels: clientLabel,
+							},
+							TopologyKey: "kubernetes.io/hostname",
+						},
+					},
+				},
+			},
+		},
+	}
+	_, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(serverPod)
```
Comment (MrHohn) [inline test/e2e/framework/util.go]: Might be good to log steps for debuggability?

File: test/e2e/framework/util.go [hunk 4]
```diff
@@ -4576,6 +4576,122 @@ func CheckConnectivityToHost(f *Framework, nodeName, podName, host string, pingC
...
+	}
+
+	for i := 0; i < 20; i++ {
+		time.Sleep(1 * time.Second)
+		resultPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(serverPod.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if resultPod.Status.ContainerStatuses[0].LastTerminationState.Terminated != nil {
```
Comment (MrHohn) [inline test/e2e/framework/util.go]: Curious: when would this pod fails? What happen if this pod starts before server pod becomes ready?

Comment (anfernee) [reply test/e2e/framework/util.go]: The server pod will crash when the bug is triggered, we don't check the client pod..

Comment (MrHohn) [reply test/e2e/framework/util.go]: Ah I see, thanks. Seems straightforward. Would probably be nice to print the logs from the pod in this case so we know it is the same connection reset error?

File: test/e2e/network/networking.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (anfernee) [reply test/e2e/network/networking.go]: Should I just remove the feature tag? What category should this be fit in?

[the remark this answers is not in the document]
Comment (MrHohn) [reply test/e2e/network/networking.go]: Per offline discussion, let's put under slow test for now. We can move it later if needed.

File: test/e2e/framework/util.go (hunk 4 above)
Comment (anfernee) [reply test/e2e/framework/util.go]: We do. but if it's something else, we cannot allow it to pass either. but I think you are right, it's better for debugging.

File: test/e2e/network/kube_proxy.go [hunk 5]
```diff
@@ -215,4 +215,116 @@ var _ = SIGDescribe("Network", func() {
...
+						},
+					},
+				},
+				RestartPolicy: v1.RestartPolicyNever,
+			},
+		}
+		_, err = fr.ClientSet.CoreV1().Pods(fr.Namespace.Name).Create(pod)
+		framework.ExpectNoError(err)
+
+		By("client pod created")
```
Comment (MrHohn) [inline test/e2e/network/kube_proxy.go]: nit: client->Client

**Comment by MrHohn** [comment]
Seems like there are some gofmt errors :)
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/76218/pull-kubernetes-verify/1114314700557914121

**Comment by MrHohn** [comment]
Merging this to keep #74840 rolling (so that we know if it fixes the issue) :)
/lgtm
/approve

**Comment by anfernee** [comment]
I have verified that this test will fail on master, but succeed if I cherry-pick #74840.

**Comment by anfernee** [comment]
Removed [Slow] gce test should fail. 
cc @danwinship

**Comment by anfernee** [comment]
The test failed as expected: `[sig-network] Network should resolve connrection reset issue #74839 ` try one more time
/retest

**Comment by anfernee** [comment]
pull-kubernetes-e2e-gce failed as expected. Adding `[Slow]` back

**Comment by MrHohn** [comment]
I guess the plan is to wait till #74840 merged first?
/kind bug
/hold

**Pull request merged**